daemon/SecretsImpl/secrets.cpp

```diff
@@ -2586,3 +2682,54 @@ void Daemon::ApiImpl::SecretsRequestQueue::handleFinishedRequest(
     }
 }
 
+void Daemon::ApiImpl::SecretsRequestQueue::dealWithDataCorruption() const
```

Comment:
Perhaps not necessary in the first implementation, but in the future it would be good if we dealt with salt data corruption differently to corruption of the test cipher data.  (In the latter case, we can recover by asking the user to re-enter the master lock code and regenerating the test cipher data.)  Perhaps add a parameter which describes the type of corruption (but is currently unused) would be useful, or add a comment regarding the possibility.

Reply:
Honestly, I believe the occourence of this code ever running is so rare that I wouldn't bother with this. But added a comment now about the possibility of doing it in the future.

I would say if a client requests further work on it, let's do this. Otherwise let's not.